packages/shared/args.js

```diff
@@ -0,0 +1,30 @@
+function args(args) {
+    const parsedArgs = {
+        named: { },
+        positional: []
+    };
+    const [...rest] = args;
+
+    rest.forEach((arg, index) => {
+        if (arg.indexOf("-") != -1) {
+            const name = getArgName(arg)
+            parsedArgs.named[name] = rest[index + 1];
+            rest.shift();
+        } else {
+            parsedArgs.positional.push(arg);
+        }
+    });
+
+    return parsedArgs;
+}
+
+function getArgName(arg) {
+    if (arg[1] === '-') return arg.slice(2, arg.length);
+    return arg.slice(1, arg.length);
+}
+
+const input = ['hello', 'world', '-c', 'config', '--filename', 'filename', 'otherargs', 'remaining'];
```

Comment:
Is your plan to handle the removal of `node cli.js` or `workflow` from `process.argv` prior to calling `args` ? If not the function needs to handle these cases.

Reply:
I think it makes sense to include the removal of the script name in this file. This could then be used in many other files. Is this simply a matter of checking for the string "node" or "workflow" and splitting as necessary?

Reply:
It should be either `node <>.js ...` or `workflow ...`, so at most two arguments. In practice it might actually just be `node <>.js ..` due to the spawn that happens [here](https://github.com/havardh/workflow/blob/master/packages/workflow-cmd/src/cli.js#L15). But I do not think we should commit to that assumption.

Reply:
I've done an implementation for this, please check if I took the right direction.